iroh/src/magicsock.rs

```diff
@@ -535,16 +535,23 @@ impl MagicSock {
                     }
                 }
 
-                let udp_pending = udp_error
-                    .as_ref()
-                    .map(|err| err.kind() == io::ErrorKind::WouldBlock)
-                    .unwrap_or_default();
-                let relay_pending = relay_error
-                    .as_ref()
-                    .map(|err| err.kind() == io::ErrorKind::WouldBlock)
-                    .unwrap_or_default();
-                if udp_pending && relay_pending {
+                let udp_pending = udp_addr.is_some()
+                    && udp_error
+                        .as_ref()
+                        .map(|err| err.kind() == io::ErrorKind::WouldBlock)
+                        .unwrap_or_default();
+
+                let relay_pending = relay_url.is_some()
+                    && relay_error
+                        .as_ref()
+                        .map(|err| err.kind() == io::ErrorKind::WouldBlock)
+                        .unwrap_or_default();
+
+                let has_path = udp_addr.is_some() || relay_url.is_some();
+
+                if udp_pending && relay_pending && has_path {
```

Comment:
Is this the right correct?

- assume relay-only path
- udp_addr is none
- so udp_pending is false
- relay_url is some
- relay_pending is true (assuming it blocked)
- has_path is true
- udp_pending (false) && .... well, doesn't matter anymore.  this is false now and it won't block?

Reply:
> - udp_addr is none
> - so udp_pending is false

Yeah - Ugh, I force-pushed a change that made this implication true (originally I had the logic correct).

Will fix.

Reply:
Updated.